dawn/src/dawn/IIR/Extents.h

```diff
@@ -116,39 +116,35 @@ class CartesianExtent : public HorizontalExtentImpl {
 
   CartesianExtent() : CartesianExtent(0, 0, 0, 0) {}
 
-  void add_impl(HorizontalExtentImpl const& other) override {
+  void addImpl(HorizontalExtentImpl const& other) override {
     auto const& otherCartesian = dynamic_cast<CartesianExtent const&>(other);
     extents_[0] += otherCartesian.extents_[0];
     extents_[1] += otherCartesian.extents_[1];
   }
 
-  void merge_impl(HorizontalExtentImpl const& other) override {
+  void mergeImpl(HorizontalExtentImpl const& other) override {
     auto const& otherCartesian = dynamic_cast<CartesianExtent const&>(other);
     extents_[0].merge(otherCartesian.extents_[0]);
     extents_[1].merge(otherCartesian.extents_[1]);
   }
 
-  void merge_impl(ast::HorizontalOffset const& offset) override {
-    auto const& offsetCartesian = ast::offset_cast<ast::CartesianOffset const&>(offset);
-    extents_[0].merge(Extent(offsetCartesian.offsetI(), offsetCartesian.offsetI()));
-    extents_[1].merge(Extent(offsetCartesian.offsetJ(), offsetCartesian.offsetJ()));
-  }
+  void addCenterImpl() override { mergeImpl(CartesianExtent()); }
 
-  bool equals_impl(HorizontalExtentImpl const& other) const override {
+  bool equalsImpl(HorizontalExtentImpl const& other) const override {
     auto const& otherCartesian = dynamic_cast<CartesianExtent const&>(other);
     return extents_[0] == otherCartesian.extents_[0] && extents_[1] == otherCartesian.extents_[1];
   }
 
-  std::unique_ptr<HorizontalExtentImpl> clone_impl() const override {
+  std::unique_ptr<HorizontalExtentImpl> cloneImpl() const override {
     return std::make_unique<CartesianExtent>(extents_[0].minus(), extents_[0].plus(),
                                              extents_[1].minus(), extents_[1].plus());
   }
 
-  bool isPointwise_impl() const override {
+  bool isPointwiseImpl() const override {
     return extents_[0].isPointwise() && extents_[1].isPointwise();
   }
 
-  std::unique_ptr<HorizontalExtentImpl> limit_impl(int minus, int plus) const override {
+  std::unique_ptr<HorizontalExtentImpl> limitImpl(int minus, int plus) const override {
```

Comment:
A final change I am going to do  here is changing `limit` to 
```
Extents::limit(Extents const& other);
```
because `minus` and `plus` don't make sense for unstructured, while when RHS is `Extents`, it makes much more sense.

Reply:
I agree. Would it make sense if `other` was an rvalue instead? It seems to me that you would never reuse your limit. Btw, I think there's only one use case for `limit()` (in cuda codegen).

Reply:
No, binding an rvalue to `const&` comes without a cost, but this allows us to also call it with an lvalue.
Regarding the other point: Yes there is only one use-case. I think it is a valid addition to the interface as it is the opposite of merge. But there is another caveat:
```
void Extent::merge(Extent const& other);
Extent Extent::limit(Extent const& other);
```

I propose symmetry:
```
void Extent::merge(Extent const& other);
void Extent::limit(Extent const& other);
Extent merge(Extent const&, Extent const&);
Extent limit(Extent const&, Extent const&);
```